python/sparkdl/estimators/keras_image_file_estimator.py

```diff
@@ -241,23 +208,30 @@ def _collectModels(self, kerasModelsBytesRDD):
         :param paramMaps: list of ParamMaps matching the maps in `kerasModelsRDD`
         :return: list of MLlib models
         """
-        transformers = []
-        for (param_map, model_bytes) in kerasModelsBytesRDD.collect():
+        # we can just use the ordering of rdd because they are in the same order as params map
+        for i, (param_map, model_bytes) in enumerate(kerasModelsBytesRDD.collect()):
```

Comment:
Do you know if this assumption always holds? IIRC Spark doesn't tend to reorder results unless necessary, but it seems like it'd be safer to create your make your original RDD an RDD of `(paramMap, paramMapIndex)` and then pass the index through the map calls.

Reply:
ah! perhaps something like the following?
```python
# somewhere
map2Index = dict(enumerate(paramMaps))
# in the for loop
i = map2Index[param_map]
```

I agree it is definitely safer and that the ordering guarantee is not explicitly documented. But I'd also be scared if spark messes this up:
```
anotherRDD = anRDD.map(someFunction)
```
link to Spark Scaladoc > Dataset > Map :
https://spark.apache.org/docs/latest/api/scala/index.html#org.apache.spark.sql.Dataset@map[U](func:org.apache.spark.api.java.function.MapFunction[T,U],encoder:org.apache.spark.sql.Encoder[U]):org.apache.spark.sql.Dataset[U]


@smurching, What are your thoughts?